### ISSUE TITLE
Bug: `JSXElement` type is not listed in `.value` of `JSXAttribute`

### ISSUE THREAD

**Issue by auvred** [typescript-eslint/typescript-eslint#8280]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&showAST=es&fileType=.tsx&code=DwQQDmAEAWD2BO8EF5gBMCWA3SB6AfHvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type value = TSESTree.JSXAttribute['value']
//   ^?
```

### Expected Result

```ts
type value = TSESTree.JSXAttribute['value']
//   ^? type value = TSESTree.JSXElement | TSESTree.JSXExpression | TSESTree.Literal | null
```

### Actual Result

```ts
type value = TSESTree.JSXAttribute['value']
//   ^? type value = TSESTree.JSXExpression | TSESTree.Literal | null
```

### Additional Info

ESLint Stylistic has `JSXAttribute > JSXElement` in the rule listener: [source code](https://github.com/eslint-stylistic/eslint-stylistic/blob/8eaf62dba81e5efebd77ef975abf1220a9b67b8b/packages/eslint-plugin-jsx/rules/jsx-curly-brace-presence/jsx-curly-brace-presence.ts#L386). this is not yet supported by `ast-spec`: 

https://github.com/typescript-eslint/typescript-eslint/blob/920f909debc3eb394f5aef4e36568bdf6641e304/packages/ast-spec/src/jsx/JSXAttribute/spec.ts#L8-L12

---

This is the code that produces `JSXAttribute > JSXElement`:

```tsx
<App horror=<div /> />
```

Ref: [eslint-stylistic tests](https://github.com/eslint-stylistic/eslint-stylistic/blob/8eaf62dba81e5efebd77ef975abf1220a9b67b8b/packages/eslint-plugin-jsx/rules/jsx-curly-brace-presence/jsx-curly-brace-presence.test.ts#L431-L434)

### Versions

| package                                | version  |
| -------------------------------------- | -------- |
| `@typescript-eslint/typescript-estree` | `6.19.0` |
| `TypeScript`                           | `5.3.3`  |

**Comment by Josh-Cena** [comment]
This was a relatively recent addition to the TS parser (though not the JSX spec) and the types aren't updated yet.

**Comment by JoshuaKGoldberg** [comment]
...huh! TIL this is allowed in TypeScript now. Sure, why not.
